File: src/main/java/com/sesame/worker/DocumentationFactoryImpl.java
package com.sesame.worker;

import com.sesame.core.worker.DocumentationWorker;
import org.springframework.stereotype.Component;

@Component
public final class DocumentationFactoryImpl implements DocumentationFactory {

    private DocumentationWorker textDocumentationWorker;

    private DocumentationWorker yamlDocumentationWorker;

    public DocumentationFactoryImpl(DocumentationWorker textDocumentationWorker,
                                    DocumentationWorker yamlDocumentationWorker) {
        this.textDocumentationWorker = textDocumentationWorker;
        this.yamlDocumentationWorker = yamlDocumentationWorker;
    }

    @Override
    public final DocumentationWorker getTextDocumentationWorker() {
        return textDocumentationWorker;
    }

    @Override
    public final DocumentationWorker getYamlDocumentationWorker() {
        return yamlDocumentationWorker;
    }

}

File: src/main/kotlin/com/sesame/core/worker/DocumentationFactoryImpl.kt
package com.sesame.core.worker

import org.springframework.stereotype.Component

@Component
class DocumentationFactoryImpl(private val textDocumentationWorker: TextDocumentationWorkerImpl,
                               private val yamlDocumentationWorker: YamlDocumentationWorkerImpl) : DocumentationFactory {

    override fun getTextDocumentationWorker(): DocumentationWorker {
        return this.textDocumentationWorker
    }

    override fun getYamlDocumentationWorker(): DocumentationWorker {
        return this.yamlDocumentationWorker
    }

}